core/src/client/lwm2m_client_core.c

```diff
@@ -373,17 +423,44 @@ static int Lwm2mCore_ObjectStoreCreateOptionalResourceHandler(void * context, Ob
 int Lwm2mCore_CreateOptionalResource(Lwm2mContextType * context, ObjectIDType objectID, ObjectInstanceIDType objectInstanceID, ResourceIDType resourceID)
 {
     ResourceDefinition * definition = Definition_LookupResourceDefinition(context->Definitions, objectID, resourceID);
-    if ((definition == NULL) || (definition->Handlers.CreateOptionalResource == NULL))
+    if (definition == NULL)
     {
-        Lwm2m_Error("No create resource handler for object ID %d resource ID %d\n", objectID, resourceID);
+        Lwm2m_Error("No resource definition for object ID %d resource ID %d\n", objectID, resourceID);
         Lwm2mResult_SetResult(Lwm2mResult_NotFound);
         goto error;
     }
 
-    if (definition->Handlers.CreateOptionalResource(context, objectID, objectInstanceID, resourceID) == 0) 
+    if (definition->Handlers.CreateOptionalResource == NULL)
```

Comment:
Could rearrange this function to match the style: if (condition_good) { do_good  } else { handle error }